core/model/src/main/java/org/eclipse/sw360/antenna/model/artifact/facts/ArtifactSoftwareHeritageID.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) Bosch Software Innovations GmbH 2019.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.sw360.antenna.model.artifact.facts;
+
+import org.eclipse.sw360.antenna.api.exceptions.AntennaInvalidArgumentException;
+import org.eclipse.sw360.antenna.model.artifact.ArtifactFact;
+import org.eclipse.sw360.antenna.model.artifact.ArtifactFactWithPayload;
+
+public class ArtifactSoftwareHeritageID extends ArtifactFactWithPayload<String> {
+
+    private ArtifactSoftwareHeritageID(String softwareHeritageID) {
+        super(softwareHeritageID);
+    }
+
+    public static class Builder {
+        private String softwareHeritageID;
+
+        public Builder(String softwareHeritageID) {
+            this.softwareHeritageID = softwareHeritageID;
+        }
+
+        public ArtifactSoftwareHeritageID build() throws AntennaInvalidArgumentException {
+            if(softwareHeritageID == null || !isValid(softwareHeritageID)) {
+                throw new AntennaInvalidArgumentException(softwareHeritageID
+                        + " does not match expected format for SoftwareHeritageID: "
+                        + getExpectedFormat());
+            }
+            return new ArtifactSoftwareHeritageID(softwareHeritageID);
+        }
+
+        private boolean isValid(String swhID) {
+            String[] swhElements = swhID.split(":");
+
+            if (swhElements.length == 4
+                    || swhElements[0].equals("swh")
+                    || isParsableInteger(swhElements[1])
+                    || "rel".equals(swhElements[2])
```

Comment:
I am not sure that "rel" is the only allowed id here, rel is definitely preferable, but I did not always find a rel, but still an id, that matched the version.